orderer/common/channelparticipation/restapi.go

```diff
@@ -254,8 +254,9 @@ func (h *HTTPHandler) sendJoinError(err error, resp http.ResponseWriter) {
 		// The client is trying to join an app-channel that exists, but the system channel does not;
 		// The client is trying to join the system-channel, and it exists. GET & DELETE are allowed on the channel.
 		h.sendResponseNotAllowed(resp, errors.WithMessage(err, "cannot join"), http.MethodGet, http.MethodDelete)
-	case types.ErrAppChannelsAlreadyExists:
+	case types.ErrAppChannelsAlreadyExists, types.ErrChannelPendingRemoval:
```

Comment:
IIUC it would not necessarily always be the server's fault. It might be something wrong with the network config. I did update both to return Conflict for pending removals and bad request for removal failures.

Reply:
I error response codes look good now, thanks.

Let me just clarify what I meant (at the risk of stating the obvious). If the "Remove" fails because of any reason that is the fault of the server, then I think it is a 5xx. In contrast, if the request caused it, or the user could have avoided it by altering the request, it is 4xx. So even if the "Remove" failed because of a networking problem in the "backend" of the server, say an NFS problem, or a cluster communication problem, it translates to 5xx. The client doesn't care that the "server" is really complex distributed system.